src/ChangeStream.php

```diff
@@ -57,16 +57,19 @@ public function getId()
 
     public function key()
     {
-        return $this->key;
+        if ($this->valid() !== null) {
```

Comment:
`valid()` returns a boolean, so this condition would always be satisfied.

Reply:
If next() is called multiple times without any changes to the collection, should we return the most recent key or should we return null for the key since valid() will return false?

Reply:
If the current element is the most recent change, `valid()` returns `true` and `key()` returns the appropriate index. If we then call `next()` and there is nothing to advance to, `valid()` should then return `false` and `key()` should return `null` (i.e. error condition, since `key()` and `current()` are not supposed to be called if `valid()` is `false`).

The `foreach` construct depends on `valid()` returning `true` and `false` to stop iterating. If we never return `false` at the end of iteration, a `foreach` loop might never terminate. This is one reason why we can't use `foreach` to iterate on a tailable `MongoDB\Driver\Cursor`, since it starts being invalid once we reach the end and we require `next()` to coerce libmongoc into checking for a next element.